### PR TITLE
[TRA-569] Add positions endpoint for vaults

### PR DESCRIPTION
**Pull request by vincentwschau**
### Changelist
- add positions endpoint for vaults
- uses factored out helper function to get subaccount fields and extract to vault response

### Test Plan
- unit tests

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new API endpoint to retrieve megavault positions at `/vault/v1/megavault/positions`.
	- Added new schemas for `VaultPosition` and `MegavaultPositionResponse` to support the new endpoint.

- **Bug Fixes**
	- Removed deprecated references to `vaultSubaccount` in the PnlTicks store tests.

- **Tests**
	- Enhanced the vault controller test suite by adding new test cases for megavault positions, ensuring robust coverage of various scenarios.

- **Documentation**
	- Updated API documentation to include the new endpoint and response structures for megavault positions.

### CODE REVIEW

File: indexer/services/comlink/src/controllers/api/v4/addresses-controller.ts [hunk 1]
```diff
@@ -75,6 +75,7 @@ class AddressesController extends Controller {
         [],
       ),
       BlockTable.getLatest(),
+      // vincent left his laptop unlocked and says hello
```
Comment (shrenujb) [inline indexer/services/comlink/src/controllers/api/v4/addresses-controller.ts:78]: Lol who was this? @tqin7?

File: indexer/services/comlink/__tests__/controllers/api/v4/vault-controller.test.ts [hunk 2]
```diff
@@ -251,5 +256,156 @@ describe('vault-controller#V4', () => {
         ]),
       );
     });
+
+    it('Get /megavault/positions with 1 vault subaccount', async () => {
```
Comment (shrenujb) [inline indexer/services/comlink/__tests__/controllers/api/v4/vault-controller.test.ts:260]: Add a test for 0 vault subaccounts and 0 positions?

File: indexer/services/comlink/src/controllers/api/v4/vault-controller.ts [hunk 3]
```diff
@@ -74,7 +105,121 @@ class VaultController extends Controller {
       .value();
 
     return {
-      vaultsPnl: groupedVaultPnlTicks,
+      vaultsPnl: _.sortBy(groupedVaultPnlTicks, 'ticker'),
+    };
+  }
+
+  @Get('/megavault/positions')
+  async getMegavaultPositions(): Promise<MegavaultPositionResponse> {
```
Comment (tqin7) [inline indexer/services/comlink/src/controllers/api/v4/vault-controller.ts:113]: this doesn't yet include `megavault` module account subaccoun 0? Is that planned for the future

Comment (vincentwschau) [reply indexer/services/comlink/src/controllers/api/v4/vault-controller.ts:113]: No, that is not part of the requirements of the endpoint as of now.

**Review by tqin7 — APPROVED** [review]

**Pull request merged**